lib/ae_mdw/sync/async_tasks/wealth_rank_accounts.ex

```diff
@@ -0,0 +1,110 @@
+defmodule AeMdw.Sync.AsyncTasks.WealthRankAccounts do
+  @moduledoc false
+
+  @typep micro_block :: term()
+
+  alias AeMdw.Collection
+  alias AeMdw.Db.Model
+  alias AeMdw.Db.State
+  alias AeMdw.Db.Mutation
+  alias AeMdw.Db.IntCallsMutation
+  alias AeMdw.Db.DeleteKeysMutation
+  alias AeMdw.Db.WriteMutation
+  alias AeMdw.Sync.Transaction
+
+  require Model
+
+  @spec dedup_pending_accounts() :: :ok
+  def dedup_pending_accounts do
+    state = State.new()
+
+    {delete_keys, pubkeys_set} =
+      state
+      |> Collection.stream(Model.AsyncTask, nil)
+      |> Stream.filter(fn {_ts, type} -> type == :store_acc_balance end)
+      |> Stream.map(&State.fetch!(state, Model.AsyncTask, &1))
+      |> Enum.map_reduce(MapSet.new(), fn
+        Model.async_task(index: index, extra_args: []), acc ->
+          {index, acc}
+
+        Model.async_task(index: index, extra_args: [extra_args]), acc ->
+          {index, MapSet.union(acc, extra_args)}
+      end)
+
+    with args when args != [] <- last_mb(state, nil) do
+      task_index = {System.system_time(), :store_acc_balance}
+      m_task = Model.async_task(index: task_index, args: args, extra_args: [pubkeys_set])
+
+      State.commit_db(
+        state,
+        [
+          DeleteKeysMutation.new(%{Model.AsyncTask => delete_keys}),
+          WriteMutation.new(Model.AsyncTask, m_task)
+        ],
+        false
+      )
+    end
+
+    :ok
+  end
+
+  @spec micro_block_accounts(micro_block(), [Mutation.t()]) :: MapSet.t()
+  def micro_block_accounts(micro_block, mutations) do
+    txs_pubkeys =
+      micro_block
+      |> :aec_blocks.txs()
+      |> Enum.flat_map(fn signed_tx ->
+        signed_tx
+        |> Transaction.get_ids_from_tx()
+        |> Enum.flat_map(fn
+          {:id, :account, pubkey} -> [pubkey]
+          _other -> []
+        end)
+      end)
+
+    txs_pubkeys
+    |> MapSet.new()
+    |> MapSet.union(int_calls_accounts(mutations))
+  end
+
+  defp int_calls_accounts(mutations) do
+    mutations
+    |> Enum.filter(&is_struct(&1, IntCallsMutation))
+    |> Enum.flat_map(fn %IntCallsMutation{int_calls: int_calls} ->
+      int_calls
+      |> Enum.filter(fn {_idx, fname, _type, _aetx, _tx} ->
+        fname in ["Chain.spend", "Call.amount"]
+      end)
+      |> Enum.flat_map(fn {_idx, _fname, _type, aetx, _tx} ->
+        aetx_accounts(aetx)
+      end)
+    end)
```

Comment:
Perhaps we can have `Sync.Block.block_mutations/4` return these addresses to avoid having to inspect so many internal data structures. Something like
```elixir  
@typep block_mutations() :: {Blocks.block_index(), block(), [Mutation.t()], [pubkey()]}
```
where the last item from the tuple would be all of the updated addresses

Reply:
Honestly I think the return of `Sync.Block.block_mutations/4` is complex enough:
```
@typep block_mutations() :: {Blocks.block_index(), block(), [Mutation.t()]}
@type height_mutations() :: {Blocks.height(), [block_mutations()]}

@spec blocks_mutations(...) :: {[height_mutations()], Txs.txi()}
```
I think it returns what is is asked for the function so I would rather filter the mutations.

Reply:
We can rename the function or return a more explicit data structure using structs. But if you want to leave it like this I won't block it